app/main/posts/collections/collections-routes.js

```diff
@@ -0,0 +1,98 @@
+module.exports = [
+    '$stateProvider',
+    function (
+        $stateProvider
+    ) {
+
+    $stateProvider
+    .state(
+        {
+            url: '^/collections/:id',
+            name: 'collectionRedirector',
+            params: {
+                id: null
+            },
+            onEnter: function ($state, collection) {
+                var viewParam = collection.view;
+                if (viewParam === 'list' || viewParam === 'data') {
```

Comment:
list is just a fallback thing for users who might have saved the wrong url when we still had a list view